packages/ra-core/src/dataFetchActions.ts

```diff
@@ -20,3 +20,28 @@ export const fetchActionsWithArrayOfRecordsResponse = [
     DELETE_MANY,
 ];
 export const fetchActionsWithTotalResponse = [GET_LIST, GET_MANY_REFERENCE];
+
+export const sanitizeFetchType = (fetchType: string) => {
```

Comment:
can't we use types for `GET_LIST` and `getList`, ... this would make switch/case/ safer.
I could try to do it in another PR

Reply:
Well, the types should ba 'open' in that case, because users may pass a custom fetch action (like `placeOrder` or `login`) and we don't want to fail at compile time in this case. 

Can be tested on another PR.

Reply:
OK I understand.